### PR TITLE
Clearer no-op logging in stubbed SavedQueryRunner

### LINKED ISSUE #9533: Clearer no-op logging for saved queries within `dbt build --include-saved-query`
This is really intended as a stubbed method that serves as a placeholder for future SL integration work. It's confusing right now that the logs of `dbt build --include-saved-query` look like:

```
10:31:43  1 of 1 START Saved Query saved_query.jaffle_shop.new_customer_orders ........... [RUN]
10:31:43  1 of 1 OK created Saved Query saved_query.jaffle_shop.new_customer_orders ...... [success in 0.00s]
```

Let's do something closer to this: https://github.com/dbt-labs/dbt-core/commit/321db7a5fa9b627f6ea42aab7d64382ee609d8a1

Screenshot 2024-02-03 at 14 56 17

### LINKED ISSUE #9629: Rename `--include-saved-query` to `--export-saved-queries`
See rationale in https://github.com/dbt-labs/dbt-core/issues/9237#issuecomment-1959861021

**Acceptance criteria:**
- Always include saved queries in the `build` command.
- Make them a `NO-OP` by default
    - https://github.com/dbt-labs/dbt-core/issues/9533
- Rename the flag to `--export-saved-queries`
- Rename the environment variable to `DBT_EXPORT_SAVED_QUERIES`
- Keep supporting the old names (`--include-saved-query`, CLI option + env var) for backwards compatibility

**Documentation to update:**
- https://docs.getdbt.com/docs/use-dbt-semantic-layer/exports#exports-in-production

### PR DESCRIPTION
**Pull request by jtcohen6**
resolves #9533
resolves #9629

### Problem

Confusing logging for `dbt build --include-saved-query`, given current stub of `SavedQueryRunner`

### Solution

Clearer logging. Needs verification with @DevonFulcher that this doesn't negatively impact SL team's downstream work. For now (unless we backport to `1.7.latest`), this only has a go-forward impact, so I'm not as worried about merging in for `main` / v1.8.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

**Review by DevonFulcher — APPROVED** [review]
LGTM! Thanks!

**Review by ChenyuLInx — APPROVED** [review]
Thanks!

**Comment by MichelleArk** [comment]
Not sure why the `Check Artifact Changes` check is failing even though there are no dbt/artifacts changes: https://github.com/dbt-labs/dbt-core/actions/runs/7980936313/job/21791611600?pr=9605

It's still pretty new, will look into that separately!

**Review by MichelleArk — APPROVED** [review]

File: core/dbt/compilation.py [hunk 1]
```diff
@@ -52,7 +52,9 @@ def print_compile_stats(stats: Dict[NodeType, int]):
         dbt.tracking.track_resource_counts(resource_counts)
 
     # do not include resource types that are not actually defined in the project
-    stat_line = ", ".join([pluralize(ct, t) for t, ct in stats.items() if ct != 0])
+    stat_line = ", ".join(
+        [pluralize(ct, t).replace("_", " ") for t, ct in stats.items() if ct != 0]
```
Comment (jtcohen6) [inline core/dbt/compilation.py:56]: I decided to undo my "fix" from https://github.com/dbt-labs/dbt-core/pull/9465, at least for the visual display in the `FoundStats` and `FinishedRunningStats` events

**Pull request merged**